### PR TITLE
Test over different neo4j versions

### PR DESCRIPTION
**Pull request by conker84**
Test over different neo4j versions with the cartesian combination of:

* Scala 2.11, 2.12
* Neo4j 3.5, 4.0, 4.1

### CODE REVIEW

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteIT.scala [hunk 1]
```diff
@@ -83,6 +85,8 @@ object SparkConnectorScalaSuiteIT {
 @Suite.SuiteClasses(Array(
   classOf[SchemaServiceTSE],
   classOf[DataSourceReaderTSE],
-  classOf[DataSourceWriterTSE]
+  classOf[DataSourceReader4xTSE],
+  classOf[DataSourceWriterTSE],
+  classOf[DataSourceWriterNeo4j4xTSE]
```
Comment (utnaf) [inline src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteIT.scala]: Can we make naming consistent? `DataSourceReaderNeo4j4xTSE`

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala [hunk 2]
```diff
@@ -83,6 +85,7 @@ object SparkConnectorScalaSuiteWithApocIT {
 @RunWith(classOf[Suite])
 @Suite.SuiteClasses(Array(
   classOf[SchemaServiceWithApocTSE],
-  classOf[DataSourceReaderWithApocTSE]
+  classOf[DataSourceReaderWithApocTSE],
+  classOf[DataSourceReader4xTSE]
```
Comment (utnaf) [inline src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala]: Shouldn't this have APOC?

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala [hunk 3]
```diff
@@ -9,12 +9,12 @@ import org.neo4j.Neo4jContainerExtension
 import org.neo4j.driver.summary.ResultSummary
 import org.neo4j.driver._
 import org.neo4j.spark.SparkConnectorScalaSuiteIT.ss
-import org.neo4j.spark.service.SchemaServiceWithApocTSE
+import org.neo4j.spark.service.{SchemaServiceTSE, SchemaServiceWithApocTSE}
```
Comment (utnaf) [inline src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala]: Is `SchemaServiceTSE` needed?

**Review by utnaf — CHANGES_REQUESTED** [review]

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala (hunk 2 above)
Comment (conker84) [reply src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala]: yes I pasted the wrong class thank you!

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala (hunk 3 above)
Comment (conker84) [reply src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteWithApocIT.scala]: done

File: src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteIT.scala (hunk 1 above)
Comment (conker84) [reply src/test/scala/org/neo4j/spark/SparkConnectorScalaSuiteIT.scala]: done

**Comment by conker84** [comment]
I'm merging this, @utnaf thank you for the feedback!

**Pull request merged**